src/document/types.ts

```diff
@@ -308,3 +308,5 @@ export type UndoRedoProcessResult<T, A extends Action, L> = {
     action: A | BaseAction;
     skip: number;
 };
+
+export type ValidationError = { message: string; details: object };
```

Comment:
It looks like `details` here has a known shape, would it not be better to define it as a type? 

From the `validateOperations` function

```
details: {
                    position: i,
                    operation: ops[i],
                    scope: ops[i].scope,
                },
 ```

Reply:
`details` is meant to be used just for log additional error info (not to be used in the code), it can be different for other errors. we can change it later if we need to consume that info in the code